design/src/rfcs/rfc0004_retry_behavior.md

```diff
@@ -0,0 +1,147 @@
+RFC: Retry Behavior
+============================
+
+> Status: RFC
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+It is not currently possible for users of the SDK to configure a client's retry behavior. This RFC establishes a method for users to set the number of retries to attempt when calling a service and would allow users to disable retries entirely.
```

Comment:
To clarify, we're only concerned about configuring max attempts for now, but using a builder pattern so that more configuration options can be added in the future in a backwards compatible manner?

Reply:
Yes. We'll be adding the ability to set an 'adaptive' retry behavior at some unspecified point in the future.